ledger/src/shred.rs

```diff
@@ -62,57 +67,39 @@ use {
         clock::Slot,
         feature_set,
         hash::{hashv, Hash},
-        packet::PACKET_DATA_SIZE,
         pubkey::Pubkey,
         signature::{Keypair, Signature, Signer},
     },
     static_assertions::const_assert_eq,
     std::fmt::Debug,
     thiserror::Error,
 };
-pub use {
-    self::{
-        legacy::{ShredCode, ShredData},
-        stats::{ProcessShredsStats, ShredFetchStats},
-    },
-    crate::shredder::Shredder,
-};
 
+mod common;
 mod legacy;
+mod shred_code;
+mod shred_data;
 mod stats;
 mod traits;
 
 pub type Nonce = u32;
+const_assert_eq!(SIZE_OF_NONCE, 4);
+pub const SIZE_OF_NONCE: usize = std::mem::size_of::<Nonce>();
 
 /// The following constants are computed by hand, and hardcoded.
 /// `test_shred_constants` ensures that the values are correct.
 /// Constants are used over lazy_static for performance reasons.
 const SIZE_OF_COMMON_SHRED_HEADER: usize = 83;
-const SIZE_OF_DATA_SHRED_HEADER: usize = 5;
-const SIZE_OF_CODING_SHRED_HEADER: usize = 6;
```

Comment:
Removing these two because  they are not used individually, and `..._HEADER` vs `..._HEADERS` is error-prone.